docs/source/primitives.rst

```diff
@@ -403,6 +403,21 @@ and `input_ref` then, if nothing is found, will fallback to building an object u
 
 .. jsonschema:: ../../src/core/schema/components/input.json
 
+
+Number Input
+-----
+
+Used to collect number information from the user. The number input component can also create or update
```

Comment:
nit: I would put the stuff about `output_ref` and `input_ref` etc in the `Input` docs and just say number input extends Input, cause I got confused reading it but didn't see anything about it explicitly on the NumberInput class